core-bundle/src/Resources/contao/classes/FrontendTemplate.php

```diff
@@ -163,6 +143,8 @@ static function ($matches)
 	 */
 	private function setCacheHeaders(Response $response)
 	{
+		// TODO: Move this. This part only depends on the current page and must also work if not using the legacy framework.
+
```

Comment:
I'll need help with this one. Where should this logic go? 

I removed the call to this method in this PR so that it isn't called when rendering legacy templates in the `AbstractFragmentController`. I could also revert the change and use `new Response($legacyTemplate->parse())` for now. But ultimately, this code has to move :slightly_smiling_face:.

Reply:
A response listener? But maybe create a new ticket for this topic?

Reply:
see https://github.com/contao/contao/issues/4403